src/test/java/com/bunq/sdk/http/PaginationScenarioTest.java

```diff
@@ -0,0 +1,114 @@
+package com.bunq.sdk.http;
+
+import com.bunq.sdk.BunqSdkTestBase;
+import com.bunq.sdk.TestConfig;
+import com.bunq.sdk.context.ApiContext;
+import com.bunq.sdk.json.BunqGsonBuilder;
+import com.bunq.sdk.model.generated.Payment;
+import com.bunq.sdk.model.generated.object.Amount;
+import com.bunq.sdk.model.generated.object.Pointer;
+import com.google.gson.Gson;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests:
+ * Pagination
+ */
+public class PaginationScenarioTest extends BunqSdkTestBase {
+
+  /**
+   * Config values.
+   */
+  private static final Integer userId = TestConfig.getUserId();
+  private static final Integer monetaryAccountId = TestConfig.getMonetaryAccountId();
+  private static final Pointer counterPartyAliasOther = TestConfig.getCounterPartyAliasOther();
+
+  private static final ApiContext apiContext = getApiContext();
+
+  /**
+   * Constants for scenario testing.
+   */
+  private final static int PAYMENT_LISTING_PAGE_SIZE = 2;
+  private final static int PAYMENT_REQUIRED_COUNT_MINIMUM = PAYMENT_LISTING_PAGE_SIZE * 2;
+  private final static int NUMBER_ZERO = 0;
+
+  /**
+   * Constants for payment creation.
+   */
+  private final static String AMOUNT_IN_EUR = "0.01";
+  private final static String FIELD_CURRENCY = "EUR";
+  private final static String FIELD_PAYMENT_DESCRIPTION = "C# test Payment";
+
+  /**
+   * Gson builder for serialization.
+   */
+  private static Gson gson = BunqGsonBuilder.buildDefault().create();
+
+  @Test
+  public void apiScenarioPaymentListingWithPaginationTest()
+  {
+    EnsureEnoughPayments();
+    Set paymentsExpected = new LinkedHashSet<>(GetPaymentsRequired());
+    Pagination paginationCountOnly = new Pagination();
+    paginationCountOnly.setCount(PAYMENT_LISTING_PAGE_SIZE);
+
+    BunqResponse<List<Payment>> paymentResponseLatest =
+        ListPayments(paginationCountOnly.getUrlParamsCountOnly());
+    Pagination paginationLatest = paymentResponseLatest.getPagination();
+    BunqResponse<List<Payment>> paymentResponsePrevious =
+        ListPayments(paginationLatest.getUrlParamsPreviousPage());
+    Pagination paginationPrevious = paymentResponsePrevious.getPagination();
+    BunqResponse<List<Payment>> paymentResponsePreviousNext =
```

Comment:
Can't this be called `paymentResponseNext`?

Reply:
But it is not next. It is next in relation to the previous. So, current. However, we care about the way we've reached it. Thus, it is `previousNext`